src/twisted/web/twcgi.py

```diff
@@ -260,45 +271,51 @@ def outReceived(self, output):
                 linebreak = delimiter[:len(delimiter)//2]
                 headers = self.headertext.split(linebreak)
                 for header in headers:
-                    br = header.find(': ')
+                    br = header.find(b': ')
                     if br == -1:
                         log.msg(
                             format='ignoring malformed CGI header: %(header)r',
                             header=header)
                     else:
                         headerName = header[:br].lower()
                         headerText = header[br+2:]
-                        if headerName == 'location':
+                        if headerName == b'location':
                             self.request.setResponseCode(http.FOUND)
-                        if headerName == 'status':
+                        if headerName == b'status':
                             try:
-                                statusNum = int(headerText[:3]) #"XXX <description>" sometimes happens.
+                                # "XXX <description>" sometimes happens.
+                                statusNum = int(headerText[:3])
                             except:
-                                log.msg( "malformed status header" )
+                                log.msg("malformed status header")
                             else:
                                 self.request.setResponseCode(statusNum)
                         else:
-                            # Don't allow the application to control these required headers.
-                            if headerName.lower() not in ('server', 'date'):
-                                self.request.responseHeaders.addRawHeader(headerName, headerText)
+                            # Don't allow the application to control
+                            # these required headers.
+                            if headerName.lower() not in (b'server', b'date'):
+                                self.request.responseHeaders.addRawHeader(
+                                    headerName, headerText)
                 output = text[headerend+len(delimiter):]
                 self.handling_headers = 0
             if self.handling_headers:
                 self.headertext = text
         if not self.handling_headers:
             self.request.write(output)
 
+
     def processEnded(self, reason):
         if reason.value.exitCode != 0:
-            log.msg("CGI %s exited with exit code %s" %
-                    (self.request.uri, reason.value.exitCode))
+            log.msg("CGI {} exited with exit code {}".format(
```

Comment:
If you're fixing up the log message formatting anyway, could you split these out into actual log message format fields?  Maybe use `twisted.logger` instead?  (As far as I know, this change is unnecessary to port to python 3, since 3.x supports %-formatting on str).